tests/e2e/basic_test.go

```diff
@@ -59,70 +79,123 @@ var _ = Suite(&BasicTestSuite{})
 	//TODO: VM2 inaccessible to this volume is currently not available
-	//c.Assert(accessible, Equals, false, Commentf("Volume %s is still available", s.volumeName))
+	//c.Assert(accessible, Equals, false, Commentf("Volume %s is still available", s.volName1))
+
+	out, err = dockercli.DeleteVolume(s.vm1, s.volName1)
+	c.Assert(err, IsNil, Commentf(out))
+
+	misc.LogTestEnd(testName, "TestBasicVolumeIsolation")
+}
+
+// Test volume isolation between _DEFAULT and user defined vmgroups:
+// Volume created in a VM belonging to _DEFAULT should be invisible
+// to VMs inside a user defined vmgroup; and vice versa.
+//
+// Test steps:
+// 1. Initialize Config DB
+// 2. Create a volume from VM1
+// 3. Verify the volume is visible to VM1 and VM2
+// 4. Create a VmGroup T1, add VM1 to T1
+// 5. Verify the volume is still visible to VM2, but invisible to VM1
+// 6. Create the 2nd volume from VM1
+// 7. Verify the 2nd volume is visible to VM1, but invisible to VM2
+// 8. Remove the volumes
+// 9. Remove Config DB
+func (s *BasicTestSuite) TestVmGroupVolumeIsolation(c *C) {
+	misc.LogTestStart(testName, "TestVmGroupVolumeIsolation")
+
+	// Initialize Config DB
+	admincli.ConfigInit(s.esx)
+
+	out, err := dockercli.CreateVolume(s.vm1, s.volName1)
+	c.Assert(err, IsNil, Commentf(out))
+
+	accessible := verification.CheckVolumeAvailability(s.vm1, s.volName1)
+	c.Assert(accessible, Equals, true, Commentf("Volume %s is not available on [%s]", s.volName1, s.vm1))
+
+	accessible = verification.CheckVolumeAvailability(s.vm2, s.volName1)
+	c.Assert(accessible, Equals, true, Commentf("Volume %s is not available on [%s]", s.volName1, s.vm2))
+
+	out, err = admincli.CreateVMgroup(s.esx, vmgroup, s.vm1Name)
+	c.Assert(err, IsNil, Commentf(out))
+
+	accessible = verification.CheckVolumeAvailability(s.vm1, s.volName1)
+	c.Assert(accessible, Equals, false, Commentf("Volume %s is still available on [%s]", s.volName1, s.vm1))
+
+	accessible = verification.CheckVolumeAvailability(s.vm2, s.volName1)
+	c.Assert(accessible, Equals, true, Commentf("Volume %s is not available on [%s]", s.volName1, s.vm2))
 
-	out, err = dockercli.DeleteVolume(vm1, s.volumeName)
+	out, err = dockercli.CreateVolume(s.vm1, s.volName2)
 	c.Assert(err, IsNil, Commentf(out))
 
-	log.Printf("END: basic-test.TestVolumeIsolation")
+	accessible = verification.CheckVolumeAvailability(s.vm1, s.volName2)
+	c.Assert(accessible, Equals, true, Commentf("Volume %s is not available on [%s]", s.volName2, s.vm1))
+
+	accessible = verification.CheckVolumeAvailability(s.vm2, s.volName2)
+	c.Assert(accessible, Equals, false, Commentf("Volume %s is still available on [%s]", s.volName2, s.vm2))
+
+	// Clean up the volumes
+	out, err = dockercli.DeleteVolume(s.vm2, s.volName1)
+	c.Assert(err, IsNil, Commentf(out))
+
+	out, err = dockercli.DeleteVolume(s.vm1, s.volName2)
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Remove Config DB
+	admincli.ConfigRemove(s.esx)
+
+	misc.LogTestEnd(testName, "TestVmGroupVolumeIsolation")
```

Comment:
> on the Dashboard have a lot of overlap and might be duplicated.

We have broke down tests into smaller steps so you might see the overlap for some steps with others. As discussed in the past we have to avoid such duplication during the automation by enhancing the existing one.

+1 on @govint's comment.